session/auth/authprovider/authconfig.go

```diff
@@ -0,0 +1,38 @@
+package authprovider
+
+type AuthConfig struct {
+	RootCAs  []string
+	KeyPairs []TLSKeyPair
+}
+
+type TLSKeyPair struct {
+	Key         string
+	Certificate string
+}
+
+type AuthTLSContextEntry struct {
```

Comment:
I'd prefer if we don't add unnecessary public API. Could you reorganize this so that `ParseRegistryAuthTLSContext` returns `map[string]*AuthConfig` directly and the intermediate public types are not needed. Also, `AuthConfig` -> `AuthTLSConfig` bit better I think.

Reply:
done